src/java/org/apache/cassandra/db/partitions/UnfilteredPartitionIterators.java

```diff
@@ -320,6 +330,17 @@ public UnfilteredRowIterator applyToPartition(UnfilteredRowIterator partition)
         return Transformation.apply(iterator, new Logging());
     }
 
+    public static void log(UnfilteredPartitionIterator partitions, String id, boolean fullDetails)
```

Comment:
should this really be blocking and consume the iterator?  In the usage this causes double reads so wouldn't it be far better to just add the transformer?

```
if (LOG_READ_RESULTS && !readCommandFinal.isDigestQuery())
                {
                    try (UnfilteredPartitionIterator i = ReadCommandVerbHandler.instance.doRead(readCommandFinal, false).makeIterator(readCommandFinal))
                    {
                        UnfilteredPartitionIterators.log(i, safeStore.commandStore().toString(), true);
                    }
                }
                return new LocalReadData(routingKeyFinal, ReadCommandVerbHandler.instance.doRead(readCommandFinal, false), readCommand);
```

Reply:
It's not guaranteed that the iterator will be fully consumed once limits and other merging is done.

Reply:
and why do we care? 

```
public class AccordInteropRead extends ReadData
{
    private static final boolean LOG_READ_RESULTS = false;
```

this is only done if that is `true` which is blocked... there are no docs so its one of those things that is likely to get dropped over time as its dead code (and our style guide is explicit about blocking dead code).  The api is dangerous to call outside of tests  and we now have to have double read logic in prod code that we hope never gets triggered...

Reply:
Because it's difficult to tell if a wrong answer is because of coordinator side filtering or because the results were never returned and it's not straightforward to log a read response.

I could get rid of the double read by using the `ReadResponse` pretty easily and convert this to a trace level log statement. Would that address it?

Reply:
to be clear, my issue isn't 100% with the double read, its the debug logic that can't be enabled polluting production code... this type of thing happened a lot in this patch and its something we should ask "do we want to maintain this"... and in every case so far my answer is "no". 

If you had a debug utility function that could take a `ReadResponse` and log it, and its dead code, I would be ok with this (we would need to document that this is a test debug utility, which is why its dead code).

Reply:
OK, I'll remove it.